src/dune_engine/action_exec.ml

```diff
@@ -516,8 +516,29 @@ let exec_until_all_deps_ready ~ectx ~eenv t =
   let+ () = loop ~eenv in
   Exec_result.{ dynamic_deps_stages = List.rev !stages }
 
-let exec ~targets ~context ~env ~rule_loc ~build_deps ~execution_parameters t =
+let _BUILD_PATH_PREFIX_MAP = "BUILD_PATH_PREFIX_MAP"
+
+let extend_build_path_prefix_map env how map =
+  let new_rules = Build_path_prefix_map.encode_map map in
+  Env.update env ~var:_BUILD_PATH_PREFIX_MAP ~f:(function
+    | None -> Some new_rules
+    | Some existing_rules ->
+      Some
+        (match how with
+        | `Existing_rules_have_precedence -> new_rules ^ ":" ^ existing_rules
```

Comment:
Looks like all call sites use `New_rules_have_precedence`, so this code path is currently dead.

Reply:
Indeed. I think having the argument is useful to force callers to think about it. We could remove the `Existing_rules_have_precedence` case, though the API feels somewhat incomplete without it, and support is just a one liner.